### DISCUSSION TITLE
API check whether OAuth Client ID/Secret are valid?

### DISCUSSION THREAD

**Discussion by mjuric** [community/community#24813] [Apps, API and Webhooks]
<p>Hi,</p>
<p>I’m developing a tool to automate configuration and deployment of an OAuth app authenticating via GitHub (JupyterHub), and I frequently see users mistype OAuth ClientId or secret when configuring auth.</p>
<p>Is there a way to programmatically verify that the OAuth Client ID and Secret are correct, so the users can be warned immediately rather than wait for post-deployment with failures? That way a command line toll we’re writing could validate the configs before deployment.</p>
<p>We can assume the user running this code has authenticated to GitHub (i.e., has a personal access token).</p>
<p>PS: it would be even better if there’s a programmatic way to register/manipulate OAuth apps?</p>
<p>Thanks!</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/mjuric">@mjuric</a>,</p>
<p>Thanks for being here and elaborating on your use case. It’s currently not possible to automate the <a href="https://developer.github.com/apps/building-oauth-apps/creating-an-oauth-app/" rel="nofollow noopener">creation of an OAuth app</a>.</p>
<p>We already received similar feature requests from other customers and I added you to the list of requesters in our internal tracking system.</p>

**Comment by dowski** [comment]
Hi, full automation aside, is there any recommendation for validating the Client ID/Secret?
